litellm/proxy/_experimental/mcp_server/mcp_server_manager.py

```
@@ -503,43 +504,64 @@ async def _fetch_tools_with_timeout(
 
         async def _list_tools_task():
             try:
-                await client.connect()
+                try:
+                    await client.connect()
+                except asyncio.CancelledError as e:
+                    verbose_logger.warning(
+                        f"Client operation cancelled for {server_name}: {str(e)}"
+                    )
+                    raise asyncio.CancelledError(
```

can we raise the raw error instead? @uc4w6c 

this will make it easier to debug on the UI why the call is failing. instead of requiring the user to get access to server logs to see what's going on. e.g. a malformed header to the mcp server.

---

ack - let me see how i can raise this